sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -54,7 +54,10 @@ private class MutableInt(var i: Int)
 
 case class Origin(
   line: Option[Int] = None,
```

Comment:
With startIndex & stopIndex, we can actually remove line & startPosition.
To make sure this can merge into 3.3, I will do the code clean up later.